### ISSUE TITLE
Keyboard users cannot interact with any part of calendar

### ISSUE THREAD

**Issue by erinchambers** [CDRH/lewisandclark#255]
The Calendar view uses generic elements (`<div>`) for interactive elements instead of real semantic links or buttons (`<a>` or `<button>`), and for this reason, keyboard navigation/tab order skips directly from the main menu to the links in the footer. 

See [Browse Calendars](https://cdrhrailsdev.unl.edu/lewisandclark/journals/calendar) and hover over an interactive date/day number. Compare that to tabbing through the page.

Error from the accessibility tree in Firefox (this is not an error that would get picked up with automated testing like axeDevtools):
<img width="437" height="94" alt="Image" src="https://github.com/user-attachments/assets/d5ee78d8-1b6c-446f-9359-c57d6ecaf532" />
